app.js

```diff
@@ -1,18 +1,84 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
+const SpotifyWebApi = require("spotify-web-api-node");
+
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) => console.log("Something went wrong when retrieving an access token", error));
 
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
+app.get("/", (req, res, next) => {
+  spotifyApi
+    .searchArtists("madonna")
+    .then((data) => {
+      // console.log(data.body);
+      res.send(data.body.artists.items[0].images);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+});
 
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.get("/homepage", (req, res) => {
+  res.render("homepage");
+});
+
+app.get("/artist-search", (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.artistName)
```

Comment:
also another way to do it its first create a variable for example
`const artistName=req.query.artistName`
and put the variable instead the full path, it would be like 
```
spotifyApi
    .searchArtists( artistName)
```